Mathlib/RingTheory/HahnSeries.lean

```diff
@@ -1270,12 +1270,9 @@ def toPowerSeriesAlg : HahnSeries ℕ A ≃ₐ[R] PowerSeries A :=
   { toPowerSeries with
     commutes' := fun r => by
       ext n
-      simp only [algebraMap_apply, PowerSeries.algebraMap_apply, C_apply,
-        coeff_toPowerSeries]
-      cases' n with n
-      · simp [PowerSeries.coeff_zero_eq_constantCoeff, single_coeff_same]
-      · simp [n.succ_ne_zero, Ne.def, not_false_iff, single_coeff_of_ne]
-        rw [PowerSeries.coeff_C, if_neg n.succ_ne_zero] }
+      simp [algebraMap_apply, PowerSeries.algebraMap_apply, PowerSeries.coeff_C, single_coeff,
```

Comment:
Here
```
ext n
      cases n <;> simp [algebraMap_apply, PowerSeries.algebraMap_apply]
```
works, and I find it more readable than digging into `Decidable` instances... what do you think?

Reply:
Yeah. I'm a bit disappointed that we need to do the case-split, but avoiding it isn't as pretty as I'd hoped.